pandas/core/arrays/datetimelike.py

```diff
@@ -437,6 +437,14 @@ def _with_freq(self, freq):
         arr._freq = freq
         return arr
 
+    def factorize(self, na_sentinel=-1):
+        if self.freq is not None:
+            # We must be unique, so can short-circuit (and retain freq)
+            codes = np.arange(len(self), dtype=np.intp)
+            # TOOD: copy?
+            return codes, self[:]
```

Comment:
I think this should copy, I would not expect that if I modify the result of `factorize`, it modifies the original array.